UiPharmecy/app/src/main/java/com/example/ahsan/uipharmecy/util/BitmapUtils.java

```diff
@@ -0,0 +1,124 @@
+package com.example.ahsan.uipharmecy.util;
+
+import android.content.ComponentName;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.database.Cursor;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.os.Parcelable;
+import android.provider.MediaStore;
+import android.system.ErrnoException;
+
+import com.example.ahsan.uipharmecy.R;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by sharma on 5/17/16.
+ */
+public class BitmapUtils {
+
+    public static boolean isUriRequiresPermissions(Uri uri, Context context) {
+        try {
+            ContentResolver resolver = context.getContentResolver();
+            InputStream stream = resolver.openInputStream(uri);
+            stream.close();
+            return false;
+        } catch (FileNotFoundException e) {
+            if (e.getCause() instanceof ErrnoException) {
+                return true;
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
@ahammad-reyad 
For API >= 23 we need to check specifically that we have permissions to read external storage,
 but we don't know if we need to for the URI .so the simplest way is to try to open the stream and see if we get an error or not. Applications are not allowed direct access to other application’s internal data.Two classes ( the ContentResolver and the ContentProvider) help to  enforce this requirement.Content Resolver  accepts requests from clients, and resolves these requests by directing them to the content provider.If we succeed in this case it will return false because in mainActivity.java we use the condition as (!boolean ) format .so ultimately it converts to true.